lib/http.ts

```diff
@@ -64,3 +66,32 @@ export function post(url: string, headers: any, data?: any): Promise<any> {
     .then(res => checkJSON(res.data))
     .catch(wrapError);
 }
+
+export function postBinary(
+  url: string,
+  headers: any,
+  data: Buffer | Readable,
+  contentType?: string,
+): Promise<Readable> {
+  return new Promise(resolve => {
+    return resolve(contentType ? contentType : detectContentType(data));
+  }).then((contentType: string) => {
+    headers["Content-Type"] = contentType;
+    headers["User-Agent"] = userAgent;
+    const responseType = "stream";
```

Comment:
It's upload, not download, and response type don't need to be `stream`. Just use `checkJSON` and return JS object.

Reply:
I understand it. 
This API returns empty response when succeeded to upload, but returns JSON when failed.
so I think I can't consistently treat response as JSON.
What is the ideal implementation? It's ok to branch by status code?

Reply:
Ah, you are right.

I guess non-200 responses are handled by `wrapError`.

For empty response,  Only file upload and DELETEs use empt response according to the official doc, so I think we can just skip `checkJSON` and return `res.data` as it is. Can you please also remove `checkJSON` from DELETE?

Reply:
thank you. ok.